packages/contracts/contracts/delegateTargets/VirtualAppAgreement.sol

```diff
@@ -12,47 +12,49 @@ contract VirtualAppAgreement {
   using Transfer for Transfer.Terms;
 
   struct Agreement {
-    Registry registry;
-    NonceRegistry nonceRegistry;
+    address registryAddr;
     Transfer.Terms terms;
+    uint256 expiry;
+    bytes32 target;
     uint256 capitalProvided;
-    uint256 loanDuration;
-    bytes32 virtualStateDeposit;
-    address intermediary;
+    address[2] beneficiaries;
   }
 
   function delegateTarget(
-    bytes32 salt,
-    uint256 nonce,
     Agreement agreement
   )
     public
   {
+    Registry registry = Registry(agreement.registryAddr);
+    require(agreement.expiry <= block.number, "agreement has not expired yet");
```

Comment:
I think the string `"agreement has not expired yet"` does not encapsulate the error state here. This is the case where someone tries to cash out their proxy agreement but the agreed amount of time for which the funds are locked up has not been reached yet; this is not really an expiry so much as a "release of borrowed / escrowed funds"